iml-agent/src/daemon_plugins/postoffice.rs

```diff
@@ -0,0 +1,173 @@
+        let routes = Arc::clone(&self.routes);
+        let inotify = Arc::clone(&self.inotify);
+        let wd = Arc::clone(&self.wd);
+        let conf_path = env::get_var("POSTMAN_CONF_PATH");
+
+        async move {
+            if let Ok(file) = fs::read_to_string(&conf_path).await {
+                let itr = file.lines().map(|mb| {
+                    let trigger = start_route(mb.to_string());
+                    (mb.to_string(), trigger)
+                });
+                routes.lock().extend(itr);
+            } else {
+                fs::OpenOptions::new()
+                    .write(true)
+                    .create(true)
+                    .open(&conf_path)
+                    .await?;
+            }
+
+            wd.lock().0 = inotify
+                .lock()
+                .add_watch(&conf_path, WatchMask::MODIFY)
+                .map_err(|e| tracing::error!("Failed to watch configuration: {}", e))
+                .ok();
+
+            let watcher = async move {
+                let mut buffer = [0; 32];
+                let mut stream = inotify.lock().event_stream(&mut buffer)?;
+
+                while let Some(event_or_error) = stream.next().await {
+                    tracing::debug!("event: {:?}", event_or_error);
+                    match fs::read_to_string(&conf_path).await {
+                        Ok(file) => {
+                            let newset: HashSet<String> =
+                                file.lines().map(|s| s.to_string()).collect();
+                            let oldset: HashSet<String> = routes.lock().keys().cloned().collect();
+
+                            let added = &newset - &oldset;
+                            let itr = added.iter().map(|mb| {
+                                let trigger = start_route(mb.to_string());
+                                (mb.to_string(), trigger)
+                            });
+                            let mut rt = routes.lock();
+                            rt.extend(itr);
+                            for rm in &oldset - &newset {
+                                if let Some(trigger) = rt.remove(&rm) {
+                                    stop_route(trigger);
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            tracing::error!("Failed to open configuration {}: {}", &conf_path, e)
+                        }
+                    }
+                }
+                tracing::debug!("Ending Inotify Listen for {}", &conf_path);
+                Ok::<_, ImlAgentError>(())
+            };
+            tokio::spawn(watcher);
+            Ok(None)
+        }
+        .boxed()
+    }
+
+    fn teardown(&mut self) -> Result<(), ImlAgentError> {
+        if let Some(wd) = self.wd.lock().0.clone() {
+            let _ = self.inotify.lock().rm_watch(wd);
+        }
+        for (_, tx) in self.routes.lock().drain() {
```

Comment:
`self.routes.lock().clear();`

`Drop` will be called on all `trigger`s when they go out of scope.

Reply:
I thought it didn't, but it turns out `systemctl stop rust-iml-agent` doesn't run `teardown()` for daemon agents

Reply:
Regardless, this code doesn't need to be calling drop explicitly.

Can you open a new issue on not calling teardown on service stop?